### PR TITLE
Vertical Scrollbar in Textarea in IE11 fix

### LINKED ISSUE #239: {{ }} hard-coded instead of using $interpolateProvider.startSymbol() and endSymbol()
In xeditable.js there are a couple of instances where the code delimiters {{ }} are hard-coded instead of using $interpolateProvider.startSymbol() and endSymbol(), which causes the code to break for  developers that have changed the default delimiters, e.g. to [[ ]]. 

They're found in the directive `editableRange` at line 180:

    this.inputEl.after('<output>{{$data}}</output>');

and the directive `editableRadiolist` at line 200:

    '<input type="radio" ng-model="$parent.$data" value="{{'+parsed.locals.valueFn+'}}">'+

### PR DESCRIPTION
**Pull request by ckosloski**
- Vertical Scrollbar in Textarea in IE11 fix
- Fixed editable-popover tests
- Updated dev-bsdate tests
- Removed code that looped through watchers.  No longer needed due to #489
- Fixes #239, #498

### CODE REVIEW

File: src/js/editable-element/controller.js [hunk 1]
```diff
@@ -313,15 +313,6 @@ angular.module('xeditable').factory('editableController',
...
-      // Manually remove the watcher on 'has-error' to prevent a memory leak on it.
-      for (var i = 0, len = $scope.$$watchers.length; i < len; i++) {
-        if ($scope.$$watchers[i] !== undefined && $scope.$$watchers[i].last && $scope.$$watchers[i].last !== undefined &&
-            typeof $scope.$$watchers[i].last === 'object' && "has-error" in $scope.$$watchers[i].last) {
-          $scope.$$watchers.splice(i, 1);
-          break;
-        }
-      }
-
       // onhide
```
Comment (eugef) [inline src/js/editable-element/controller.js]: Why did you remove this code? Is it not necessary?

Comment (ckosloski) [inline src/js/editable-element/controller.js]: I had originally added it and it is no longer needed now that #489 has been merged.

Comment (eugef) [inline src/js/editable-element/controller.js]: Ok.

**Comment by ckosloski** [comment]
@eugef I meant to say #439, not #239.  Can you reopen #239 and close #439?

**Comment by artemgrygor** [comment]
@vitalets @eugef Guys, when do u plan to publish it at NPM?

**Comment by Burnett01** [comment]
Not sure what happened here, but the code removed by @ckosloski  is still available:

Removed:  https://github.com/vitalets/angular-xeditable/pull/500/files#r70768015

Still there:  https://github.com/vitalets/angular-xeditable/blob/master/dist/js/xeditable.js#L798

---

This needs to be addressed asap. I just ran into an issue where a scope watcher **did not exist**, yet the loop was called and it failed miserably. 

<pre>angular.min.js:89 TypeError: Cannot read property 'length' of null
  at EditableController.self.hide (http://localhost:3000/javascripts/libs/xeditable.min.js:801:48)
  at http://localhost:3000/javascripts/libs/xeditable.min.js:1326:18
  at Array.forEach (native)
  at Object.q [as forEach] (http://localhost:3000/javascripts/libs/angular.min.js:7:280)
  at Object.$hide (http://localhost:3000/javascripts/libs/xeditable.min.js:1325:15)
  at Object.$cancel (http://localhost:3000/javascripts/libs/xeditable.min.js:1350:12)
  at http://localhost:3000/javascripts/libs/xeditable.min.js:1190:50
  at Array.forEach (native)
  at Object.q [as forEach] (http://localhost:3000/javascripts/libs/angular.min.js:7:280)</pre>

If you pretend keeping the loop @ "https://github.com/vitalets/angular-xeditable/blob/master/dist/js/xeditable.js#L798", you **must** make sure to only call it when the $$scope.watchers exist: 

```javascript
// Check if a watcher exists
if ($scope.$$watchers != null) {
  // Manually remove the watcher on 'has-error' to prevent a memory leak on it.
  for (var i = 0, len = $scope.$$watchers.length; i < len; i++) {
    if ($scope.$$watchers[i] !== undefined && $scope.$$watchers[i].last && $scope.$$watchers[i].last !== undefined &&
      typeof $scope.$$watchers[i].last === 'object' && "has-error" in $scope.$$watchers[i].last) {
    $scope.$$watchers.splice(i, 1);
    break;
    }
  }
}
```

@eugef

**Comment by artemgrygor** [comment]
It's impossible to use package with such error! @eugef @vitalets @atodicebear

**Comment by davicente** [comment]
Please, could you add @Burnett01 correction??    

> if ($scope.$$watchers != null) {

It 's blocking

**Pull request merged**